crates/router/src/core/payments/helpers.rs

```diff
@@ -5445,13 +5445,13 @@ pub async fn get_unified_translation(
 }
 pub fn validate_order_details_amount(
     order_details: Vec<api_models::payments::OrderDetailsWithAmount>,
-    amount: i64,
+    amount: MinorUnit,
     should_validate: bool,
 ) -> Result<(), errors::ApiErrorResponse> {
     if should_validate {
-        let total_order_details_amount: i64 = order_details
+        let total_order_details_amount: MinorUnit = order_details
             .iter()
-            .map(|order| order.amount * i64::from(order.quantity))
+            .map(|order| order.amount * MinorUnit::new(i64::from(order.quantity)))
```

Comment:
We shouldn't be constructing a `MinorUnit` from order quantity. 

I think allowing a `MinorUnit` to be multiplied by an integer is a fair use case, you should be able to add an `impl Mul<Rhs = i64> for MinorUnit` implementation. We can even use a `u16` / `u32` instead of `i64` in the implementation, since we are sure that quantities can't be negative.

Reply:
Sure, I have refactored the Mul implementation to allow multiplying u16 by MinorUnit for use cases where we need to multiply the amount by the quantity